### ISSUE TITLE
LDAPPasswordIdentityProvider not working in release-3.10

### ISSUE THREAD

**Issue by nhenderson** [openshift/openshift-ansible#9842]
#### Description
Commit 78d6c021cb08af0b510c42725a511fe826dea930 is throwing the following error for me on install of OpenShift Origin 3.10. 

##### Version

`OpenShift Origin 3.10`

* Your ansible version per `ansible --version`
```bash 
ansible --version
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/user/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```
If you're operating from a **git clone**:

* The output of `git describe`

`openshift-ansible-3.10.41-1-22-g4c47080`

##### Steps To Reproduce
1. Set `openshift_master_identity_providers` variable:
```
openshift_master_identity_providers: [{'name': 'my_ldap_provider', 'challenge': 'true', 'login': 'true', 'kind': 'LDAPPasswordIdentityProvider', 'attributes': {'id': ['dn'], 'email': ['mail'], 'name': ['cn'], 'preferredUsername': ['uid']}, 'bindDN': '', 'bindPassword': '', 'ca': '', 'insecure': 'false', 'url': 'ldap://ldap.example.com:389/ou=users,dc=example,dc=com?uid'}] 
```
Run the install `playbooks/aws/openshift-cluster/install.yml` playbook

##### Expected Results
No errors

##### Observed Results
Error on `name` attribute key in `roles/lib_utils/filter_plugins/openshift_master.py` line 161
```
TASK [openshift_control_plane : set_fact] *****************************************************************************************************************************
task path: /var/www/html/openshift-ansible/roles/openshift_control_plane/tasks/main.yml:124
Thursday 30 August 2018  12:27:46 -0400 (0:00:01.026)       0:03:20.654 ******* 
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 139, in run
    res = self._execute()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 510, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/task.py", line 260, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/base.py", line 361, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 523, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 479, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 722, in do_template
    res = j2_concat(rf)
  File "<template>", line 12, in root
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 481, in translate_idps
    idp_inst = idp_class(api_version, idp) if idp_class is not None else IdentityProviderBase(api_version, idp)
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 161, in __init__
    self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(self._idp['name'])
KeyError: 'name'

fatal: [ip-10-209-77-220.ec2.internal]: FAILED! => {
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 139, in run
    res = self._execute()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 510, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/task.py", line 260, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/base.py", line 361, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 523, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 479, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 722, in do_template
    res = j2_concat(rf)
  File "<template>", line 12, in root
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 481, in translate_idps
    idp_inst = idp_class(api_version, idp) if idp_class is not None else IdentityProviderBase(api_version, idp)
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 161, in __init__
    self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(self._idp['name'])
KeyError: 'name'

fatal: [ip-10-209-65-92.ec2.internal]: FAILED! => {
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 139, in run
    res = self._execute()
  File "/usr/lib/python2.7/dist-packages/ansible/executor/task_executor.py", line 510, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/task.py", line 260, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/lib/python2.7/dist-packages/ansible/playbook/base.py", line 361, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 523, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 479, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/dist-packages/ansible/template/__init__.py", line 722, in do_template
    res = j2_concat(rf)
  File "<template>", line 12, in root
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 481, in translate_idps
    idp_inst = idp_class(api_version, idp) if idp_class is not None else IdentityProviderBase(api_version, idp)
  File "/var/www/html/openshift-ansible/roles/lib_utils/filter_plugins/openshift_master.py", line 161, in __init__
    self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(self._idp['name'])
KeyError: 'name'

fatal: [ip-10-209-70-160.ec2.internal]: FAILED! => {
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
```

##### Additional Information

Line 161: `self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(self._idp['name'])` in `roles/lib_utils/filter_plugins/openshift_master.py` is throwing  the error. 
I have the `name` key defined in the `openshift_master_identity_providers` variable for `kind: LDAPPasswordIdentityProvider`, yet `name` doesn't appear to be set by the time line 161 runs.

**Comment by nagonzalez** [comment]
the `_idp` dictionary is an internal copy of the identity provider. `idp` contains the original values.

I was able to get this working by replacing all instances of `self._idp['name'] ` with `idp['name'] `


for example:
```
self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(self._idp['name'])
->
self._idp['ca'] = '/etc/origin/master/{}_ldap_ca.crt'.format(idp['name'])
```

**Comment by vrutkovs** [comment]
Thanks, good catch. Created a PR to master to fix this, this would be applied to 3.10 branch later on

**Comment by vrutkovs** [comment]
Filed https://bugzilla.redhat.com/show_bug.cgi?id=1633137 to describe new behaviour in the docs
